src/component/segments/EditSegment/EditSegment.tsx

```diff
@@ -0,0 +1,99 @@
+import { CreateButton } from 'component/common/CreateButton/CreateButton';
+import FormTemplate from 'component/common/FormTemplate/FormTemplate';
+import { ADMIN } from 'component/providers/AccessProvider/permissions';
+import { useSegmentsApi } from 'hooks/api/actions/useSegmentsApi/useSegmentsApi';
+import { useConstraintsValidation } from 'hooks/api/getters/useConstraintsValidation/useConstraintsValidation';
+import { useSegments } from 'hooks/api/getters/useSegments/useSegments';
+import useUiConfig from 'hooks/api/getters/useUiConfig/useUiConfig';
+import useToast from 'hooks/useToast';
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+import { formatUnknownError } from 'utils/format-unknown-error';
+import { useSegmentForm } from '../hooks/useSegmentForm';
+import { SegmentForm } from '../SegmentForm/SegmentForm';
+
+export const EditSegment = () => {
+    const { uiConfig } = useUiConfig();
+    const { setToastData, setToastApiError } = useToast();
+    const history = useHistory();
+    const { createSegment, loading } = useSegmentsApi();
+    const { refetchSegments } = useSegments();
+    const {
+        name,
+        setName,
+        description,
+        setDescription,
+        constraints,
+        setConstraints,
+        getSegmentPayload,
+        errors,
+        clearErrors,
+    } = useSegmentForm();
+
+    const hasValidConstraints = useConstraintsValidation(
+        'default',
+        'test',
+        constraints
+    );
+
+    const formatApiCode = () => {
+        return `curl --location --request PUT '${
+            uiConfig.unleashUrl
+        }/api/admin/segments/${id}' \\
+--header 'Authorization: INSERT_API_KEY' \\
+--header 'Content-Type: application/json' \\
+--data-raw '${JSON.stringify(getSegmentPayload(), undefined, 2)}'`;
+    };
+    const handleSubmit = async (e: React.FormEvent) => {
+        e.preventDefault();
+        clearErrors();
+        try {
+            await createSegment(getSegmentPayload());
+            refetchSegments();
+            history.push('/segments/');
+            setToastData({
+                title: 'Segment created',
+                text: 'Segment created successfully created',
+                confetti: true,
+                type: 'success',
+            });
+        } catch (error: unknown) {
+            setToastApiError(formatUnknownError(error));
+        }
+    };
+    const handleCancel = () => {
+        history.goBack();
+    };
+
+    return (
+        <FormTemplate
+            loading={loading}
+            title="Create segment"
+            description="Segment makes it easy for you to define who should be exposed to your feature.
+            The segment is often a collection of constraints and can be reused.
+            Create a segment and use it when you configure an activation strategy on a feature toggle. "
+            documentationLink="https://docs.getunleash.io/how-to/how-to-create-and-assign-custom-project-roles"
```

Comment:
@FredrikOseberg 

I know this is beyond the scope of this PR, but I think we should update the form template component to make the "Learn more" link say something more meaningful:

https://www.a11yproject.com/posts/creating-valid-and-accessible-links/#write-helpful-link-text

A simple solution would be to add "about X" after "Learn more". If it's an issue that the link text gets longer and more unpredictable, we could also hide the extra text from rendering by using a `visually-hidden` class or something, but it'd be nice to have it there.

Reply:
Good idea! I'll add an optional custom docs link label and use it in the segment forms, at least.